packages/gatsby-transformer-sharp/src/create-resolvers.js

```diff
@@ -0,0 +1,18 @@
+module.exports = ({ createResolvers, reporter }) => {
+  const resolvers = {
+    File: {
+      childImageSharp: {
+        resolve: (parent, args, context, info) => {
+          if (parent.extension === `gif`) {
```

Comment:
Should we warn for every extension that we don't support. Basically using https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sharp/src/on-node-create.js#L1-L8

Reply:
@wardpeet I wanted to make this specific to .gif as I mostly seen this as an user error. Do you think it would be valuable to make this a generic error?

Reply:
What about converting it to use `${parent.extension}` that will give you the same message but for all extensions.

```
`You can't use childImageSharp together with .${parent.extension} files. The childImageSharp portion of the query in this file will return null:\n${context.componentPath}\n\nIf you want to display ${parent.name}.${parent.extension} on your site, use publicURL and a normal img tag instead.\nIf the files can be in multiple formats, you could check for a falsy value on childImageSharp and conditionally use e.g. gatsby-image.`
```

Reply:
Yeah, sure 👍

Reply:
@wardpeet Added it